resources/page/permalinks_test.go

```diff
@@ -46,6 +46,8 @@ var testdataPermalinks = []struct {
 	{"/:sections[0]/:sections[last]/", true, "/a/c/"},               // Sections
 	{"/\\:filename", true, "/:filename"},                            // Escape sequence
 	{"/special\\::slug/", true, "/special:the-slug/"},               // Escape sequence
+	{"/:contentbasename/", true, "/index/"},                         // Content base name
+	{"/:contentbasenameorslug/", true, "/index/"},                   // Content base name or slug
```

Comment:
I noticed that, for the test files, the `component` is not defined for paths, meaning that here we just get the filename (without extension) instead of the parent file's name as we'd get inside the `content` directory.